packages/toolkit/src/entities/state_selectors.ts

```diff
@@ -20,7 +25,9 @@ export function createSelectorsFactory<T, Id extends EntityId>() {
     selectState?: (state: V) => EntityState<T, Id>,
     options: GetSelectorsOptions = {}
   ): EntitySelectors<T, any, Id> {
-    const { createSelector = createDraftSafeSelector } = options
+    let { createSelector } = options
+    createSelector ??= createDraftSafeSelector
```

Comment:
for some reason typescript infers differently doing this vs the default assignment

with the default assignment it ends up as a union of the two types, rather than keeping the more generic type
![image](https://github.com/reduxjs/redux-toolkit/assets/18308300/ca6da7e7-c122-4b35-a957-3900708ddd9f)

the alternative would be 
```suggestion
    const { createSelector = createDraftSafeSelector as AnyCreateSelectorFunction } = options
```
i suppose